docs/core/diagnostics/event-counter-perf.md

```diff
@@ -0,0 +1,606 @@
+
+:::code language="csharp" source="snippets/EventCounters/MinimalEventCounterSource.cs":::
+
+The <xref:System.Diagnostics.Tracing.EventSource.WriteEvent%2A?displayProperty=nameWithType> line is the <xref:System.Diagnostics.Tracing.EventSource> part and is not part of <xref:System.Diagnostics.Tracing.EventCounter>, it was written to show that you can log a message together with the event counter.
+
+## Add an action filter
+
+The sample source code is an ASP.NET Core project. You can add an [action filter](/aspnet/core/mvc/controllers/filters#action-filters) globally that will log the total request time. Create a new file named *LogRequestTimeFilterAttribute.cs*, and used the C# code below:
+
+```csharp
+using System.Diagnostics;
+using Microsoft.AspNetCore.Http.Extensions;
+using Microsoft.AspNetCore.Mvc.Filters;
+
+namespace DiagnosticScenarios
+{
+    public class LogRequestTimeFilterAttribute : ActionFilterAttribute
+    {
+        readonly Stopwatch _stopwatch = new Stopwatch();
+
+        public override void OnActionExecuting(ActionExecutingContext context) => _stopwatch.Start();
+
+        public override void OnActionExecuted(ActionExecutedContext context)
+        {
+            _stopwatch.Stop();
+
+            MinimalEventCounterSource.Log.Request(
+                context.HttpContext.Request.GetDisplayUrl(), _stopwatch.ElapsedMilliseconds);
+        }
+    }
+}
+```
+
+The action filter starts a <xref:System.Diagnostics.Stopwatch> as the request begins, and stops after it has completed, capturing the elapsed time. The total milliseconds is logged to the `MinimalEventCounterSource` singleton instance. In order for this filter to be applied, you need to add it to the filters collection. In the *Startup.cs* file, update the `ConfigureServices` method in include this filter.
+
+```csharp
+public void ConfigureServices(IServiceCollection services) =>
+    services.AddControllers(options => options.Filters.Add<LogRequestTimeFilterAttribute>())
+            .AddNewtonsoftJson();
+```
+
+## Monitor event counter
+
+You logged the metric to the <xref:System.Diagnostics.Tracing.EventCounter>, but unless you access the statistics from of it, it is not useful. To get the statistics, you need to enable the <xref:System.Diagnostics.Tracing.EventCounter> by creating a timer that fires as frequently as you want the events, as well as a listener to capture the events. To do that, you can use [dotnet-counters](dotnet-counters.md).
+
+Use the [dotnet-counters ps](dotnet-counters.md#dotnet-counters-ps) command, to display a list of .NET processes that can be monitored.
+
+```console
+dotnet-counters ps
```

Comment:
This command won't show anything because the app isn't running -- the directions had us run and stop it.

Reply:
Also should clarify if you have to run it using dotnet run, otherwise if you run it with VS you don't get a dotnet process, you get iisexpress. In that case you'll need two command prompts open.